app/events/views.py

```diff
@@ -18,96 +19,235 @@ def post(self, user_id):
+            event.save_event()
+            response = {
+                'id': event.id,
+                'name': event.name,
+                'description': event.description,
+                'category': event.category,
+                'date': event.date,
+                'author': event.author,
+                'location': event.location,
+                'message': 'event successfully created'
+            }
+            return make_response(jsonify(response)), 201
+        return make_response(jsonify({'message': "Name cannot be blank"}))
 
     @login_required
-    def get(self, user_id, _id=None):
+    def get(self, user_id, id=None):
         """Gets the created events."""
-        events = Event.query.filter_by(author=user_id).first()
-        if not _id:
+        if id is None:
             try:
-                max_per_page = int(requests.args.get('max_per_page'))
-                page = int(request.args.get('page'))
-            except Exception as e:
-                max_per_page = 6
-                page = 1
-            events = Event.query.filter_by(author=user_id).paginate(int(page), int(max_per_page), False)
-            if events.has_prev:
-                prev_page='/events,?max_per_page={}&page={}'.format(max_per_page, events.prev_num)
-            if events.has_next:
-                next_page='/events/?max_per_page={}&page={}'.format(max_per_page, events.next_num)
-            if not events:
+                limit = int(request.args.get('limit', default=20, type=int))
+                page = int(request.args.get('page', default=1, type=int))
+            except TypeError:
+                return make_response(
+                        jsonify({"error": "limit and page must be int"})), 400
+            if int(limit) > 6:
+                limit = 6
+            else:
+                limit = int(limit)
+            q = request.args.get('q', type=str)
+            if q:
+                event = Event.query.filter_by(author=user_id).first()
+                if not event:
+                    response = {'message': 'Event is not available'}
+                    return make_response(jsonify(response)), 404
+                events = Event.query.filter_by(author=user_id).filter(
+                    Event.name.ilike(q))
+                results = []
+                for event in events:
+                    event_search = {
+                        'id': event.id,
+                        'name': event.name,
+                        'description': event.description,
+                        'category': event.category,
+                        'date': event.date,
+                        'author': event.author,
+                        'location': event.location,
+                    }
+
+                    results.append(event_search)
+                return make_response(jsonify(results=results)), 200
+            elif q:
+                events = Event.query.filter_by(author=user_id).filter(
```

Comment:
This code block is similar as the one at the top, it's better to combine the two to avoid repeating your self